db/music.dbml

```diff
@@ -0,0 +1,299 @@
+}
+
+Table genre {
+  genre_id uuid [pk, default: `gen_random_uuid()`]
+  name text [not null, unique]
+  parent_id text [null, ref: > genre.genre_id]
+
+  indexes {
+    parent_id
+  }
+}
+
+Table song_genre {
+  song_genre_id uuid [pk, default: `gen_random_uuid()`]
+  song_id int [not null, ref: > song.song_id]
+  genre_id int [not null, ref: > genre.genre_id]
+
+  indexes {
+    (song_id, genre_id) [unique]
+  }
+}
+
+Table bracket {
+  bracket_id uuid [pk, default: `gen_random_uuid()`]
+  epoch int
+  cycle int
+  name text
+  type text [not null, Note: 'year theme aggregate' ]
+  year_min int [null]
+  year_max int [null]
+  order int [not null]
+  size text [not null, Note: '128 96 64 48 32' ]
+  date_start timestamp [null]
+  date_end timestamp [null]
+  discord_message_id text
+}
+
+Table project {
+  project_id uuid [pk, default: `gen_random_uuid()`]
+  name text [unique]
+  project_lead_id int [null, ref: > participant.participant_id]
+}
+
+Table bracket_selection {
+  bracket_selection_id uuid [pk, default: `gen_random_uuid()`]
+  project_id int [not null, ref: > project.project_id]
+  epoch int [not null] 
+  cycle int [not null]
+  bracket_id int [not null, ref: > bracket.bracket_id]
+  date_selected timestamp [not null]
+}
+
+Table seed {
+  seed_id uuid [pk, default: `gen_random_uuid()`]
+  bracket_id int [not null, ref: > bracket.bracket_id]
+  seed int [not null]
+  submission_id int [not null, ref: > submission.submission_id]
+  emoji text [null]
+
+  indexes {
+    (bracket_id, seed) [unique]
+    submission_id
+  }
+}
+
+Table seed_history {
+  seed_history_id uuid [pk, default: `gen_random_uuid()`]
+  date_changed timestamp [not null]
+  reason_changed text [not null, Note: '"missed dupe" ineligible' ]
+  seed_id int [not null, ref: > seed.seed_id]
+  submission_id int [not null, ref: > submission.submission_id]
+  emoji text [null]
+  
+  indexes {
+    seed_id
+    submission_id
+  }
+}
+
+Table round {
+  round_id uuid [pk, default: `gen_random_uuid()`]
+  bracket_id int [not null, ref: > bracket.bracket_id]
+  name text [not null, Note: 'Finals "Third Place" "Semi-Finals" "Quarter-Finals" "Round of 16" "Round of 32" "Round of 64" "Round of 128" "Play-Ins"']
+  size int [not null, Note: '0 means play-in round']
+  index_asc int [not null, Note: '0-indexed from Play-Ins']
+  index_desc int [not null, Note: '0-indexed from finals']
+  date_start timestamp [not null]
+  date_end timestamp [null]
+  discord_message_id text
+
+  Indexes {
+    bracket_id
+  }
+}
+
+/*                 0idx   128/96  64/48   32
+  Finals            0        8      7      6
+  "Third Place"     1        7      6      5
+  "Semi-Finals"     2        6      5      4
+  "Quarter-Finals"  3        5      4      3
+  "Round of 16"     4        4      3      2
+  "Round of 32"     5        3      2      1
+  "Round of 64"     6        2      1      
+  "Round of 128"    7        1             
```

Comment:
This slightly reinforces my "playins should be -1" comment above as then we'd have a symmetry between the "real" rounds in the two index columns, e.g. finals could be 0 or 7 and first round (of 128) could be 0 or 7.

Reply:
Okie doke.